core/domain/question_domain_test.py

```diff
@@ -161,3 +163,174 @@ def test_update_methods(self):
 
         question.update_language_code('es')
         self.assertEqual(question.language_code, 'es')
+
+    def test_add_skill(self):
+        """Test to verify add_skill."""
+        collection_id = 'col1'
+        exp_id = '0_exploration_id'
+        owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        # Create a new collection and exploration.
+        self.save_new_valid_collection(
+            collection_id, owner_id, exploration_id=exp_id)
+
+        # Add a skill.
+        collection_services.update_collection(
+            owner_id, collection_id, [{
+                'cmd': collection_domain.CMD_ADD_COLLECTION_SKILL,
+                'name': 'test'
+            }], 'Add a new skill')
+
+        state = exp_domain.State.create_default_state('ABC')
+        question_data = state.to_dict()
+
+        question_dict = {
+            'question_id': 'col1.random',
+            'title': 'abc',
+            'question_data': question_data,
+            'question_data_schema_version': 1,
+            'collection_id': 'col1',
+            'language_code': 'en'
+        }
+
+        question = question_domain.Question.from_dict(question_dict)
+
+        question.add_skill('test', owner_id)
```

Comment:
It would be better to use a more indicative string like 'skill0' here. Ditto below.

Reply:
Updated.